crates/devolutions-pedm-shell-ext/src/lib_win.rs

```diff
@@ -288,10 +301,22 @@ enum ChannelCommand {
     Elevate(PathBuf),
 }
 
+static mut DLL_MODULE: HINSTANCE = HINSTANCE(std::ptr::null_mut());
```

Comment:
suggestion: I think the benefit of using a `static mut` is negligible over using a synchronized primitive such as `Mutex` and remove the unsafe blocks related to reading and writing to a static mut variable.

Reply:
question: why synchronize access when it's not needed in this case? Is it just so we don't need the `unsafe` blocks? If that's the case, I'm happy to change it, but I wonder if there's more reasoning or a more suitable primitive.

Reply:
It’s just so that we don’t have to allocate brain power to verifying that the code is valid (= to remove the unsafe blocks to review and to audit in case of problem).
In practice, the cost of the synchronisation here is just an atomic operation to check that the lock is not held by anything else, there is no syscall unless the synchronisation was actually necessary. Since this is not a busy loop, it’s practically free and I wouldn’t bother with unsafe code for so little. 

Typically https://doc.rust-lang.org/beta/std/sync/struct.LazyLock.html or https://doc.rust-lang.org/beta/std/sync/struct.OnceLock.html would be more appropriate primitive (initialize once, and use many times), but the cost is the same (an atomic operation to check that the value is initialized).

Alternatively, if the value is initialized and used on the same thread, you can use a thread local variable and a Cell:
- https://doc.rust-lang.org/std/macro.thread_local.html
- https://doc.rust-lang.org/std/cell/struct.Cell.html

This way there is no synchronization.
(There are also LazyCell and OnceCell.)

Reply:
Ok, maybe you can short circuit some research for me here :) 

Actually `OnceLock` looks like a good fit, but regardless I cannot use it or any `Mutex` because the underlying type of `HMODULE` / `HINSTANCE` is `c_void`.

At least, the only way(s) I can see to make it work, also use unsafe code....

Reply:
Oh I see. It would have worked if the windows crate implemented Sync and Send on HMODULE, but I understand it does not. In that case you may as well keep the current code!

Reply:
😞